packages/edit-site/src/store/private-actions.js

```diff
@@ -49,3 +53,70 @@ export const setEditorCanvasContainerView =
 			view,
 		} );
 	};
+
+/**
+ * Action that removes an array of templates.
+ *
+ * @param {Array} templates An array of template objects to remove.
+ */
+export const removeTemplates =
+	( templates ) =>
+	async ( { registry } ) => {
+		try {
+			await Promise.allSettled(
```

Comment:
The point of using `allSettled` is to inspect the return value which is of type:
```
[
  { status: 'fulfilled', value: 99 },
  { status: 'rejected', reason: Error: an error }
]
```

So the logic below for showing the messages is not right.

Reply:
I agree with this. Actions are weird, since `deleteEntityRecord` is a thunk, and adding the `async` keyword makes a difference. We should definitely be inspecting the results of `allSettled`.

Reply:
Okay, after investigating more even if `deleteEntityRecord` is a thunk, the returned result is a Promise. My initial tests where wrong by throwing inside the `map` callback. 

So what we need to do here is just probably get the result of `allSettled` and inspect the results.

Reply:
I updated the have the error logic based on allSettled result. It seems to be working well when I forced some delete errors on the backend.